config/internal/configsource/manager.go

```diff
@@ -91,6 +100,21 @@ import (
 //      # as a string if params doesn't specify that "binary" is true.
 //      text_from_file: $file:/etc/text.txt
 //
+// Bracketed single-line should be used when concatenating a suffix to the value retrieved by
+// the config source. Example:
+//
+//    component:
+//      # Retrieves the value of the environment variable LOGS_DIR and appends /component.log to it.
+//      log_file_fullname: ${env:LOGS_DIR}/component.log
+//
+// Environment variables are expanded before passed to the config source when used in the selector or
+// the optional parameters. Example:
+//
+//    component:
+//      # Retrieves the value from the file text.txt located on the path specified by the environment
+//      # variable DATA_PATH.
+//      text_from_file: $file:$DATA_PATH/text.txt
```

Comment:
In this example how do we know we need to stop parsing DATA_PATH env variable name when we hit `/`? Do we only allow a limited set of characters in env variable names?

Reply:
Yes, this follows the same logic as env var expansion done via os.ExpandEnv and Unix shell: it treats all alpha num and underscores as part of the environment variable name.

Reply:
Done: updated the text trying to clarify.